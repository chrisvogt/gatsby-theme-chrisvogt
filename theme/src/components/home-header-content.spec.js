import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import HomeHeaderContent from './home-header-content'

describe('HomeHeaderContent', () => {
  it('renders the header content correctly', () => {
    render(<HomeHeaderContent />)

    // Check for the main headline
    expect(screen.getByRole('heading', { level: 1 })).toHaveTextContent("Hi! ðŸ‘‹ I'm Chris Vogt.")

    // Check for the paragraphs
    expect(screen.getByText(/This is my personal blog and digital garden/i)).toBeInTheDocument()
    expect(screen.getByText(/I work as a software engineer at GoDaddy/i)).toBeInTheDocument()

    // Use a more flexible text matcher for the piano paragraph
    const pianoParagraph = screen.getByText(
      (content, element) =>
        element.tagName.toLowerCase() === 'p' && content.includes('Most evenings') && content.includes('piano')
    )
    expect(pianoParagraph).toBeInTheDocument()

    expect(screen.getByText(/This space is always evolving/i)).toBeInTheDocument()
  })

  it('applies wobble animation to emoji on mouse enter', () => {
    render(<HomeHeaderContent />)

    const emoji = screen.getByText('ðŸ‘‹')

    // Simulate mouse enter
    fireEvent.mouseEnter(emoji)

    // Check if the animation is applied
    expect(emoji).toHaveStyle('animation: wobble 1s ease-in-out')
  })

  it('removes wobble animation from emoji on animation end', () => {
    render(<HomeHeaderContent />)

    const emoji = screen.getByText('ðŸ‘‹')

    // Simulate mouse enter and animation end
    fireEvent.mouseEnter(emoji)
    fireEvent.animationEnd(emoji)

    // Check if the animation is removed
    expect(emoji).toHaveStyle('animation: none')
  })
})
