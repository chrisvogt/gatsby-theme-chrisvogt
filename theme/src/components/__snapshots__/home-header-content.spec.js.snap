// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`HomeHeaderContent matches the snapshot 1`] = `
<div
  className="css-1ip9uq4"
>
  <h1
    className="css-8pnozc"
    onMouseEnter={[Function]}
  >
    Welcome!
     
    <span
      className="emoji"
      onAnimationEnd={[Function]}
    >
      ðŸ‘‹
    </span>
     
    I'm Your Name.
  </h1>
  <p
    className="css-1avyp1d"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
  </p>
  <p
    className="css-1avyp1d"
  >
    Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  </p>
</div>
`;
